nncf/api/compression.py

```diff
@@ -127,7 +128,7 @@ class CompressionStage(IntEnum):
     PARTIALLY_COMPRESSED = 1
     FULLY_COMPRESSED = 2
 
-    def __add__(self, other: "CompressionStage") -> "CompressionStage":
+    def __add__(self, other: int) -> CompressionStage:
```

Comment:
I understand that `CompressionStage` inherits from `int`, but now the type hint is confusing.

Reply:
Yes, it's problem exists, but
```
nncf/api/compression.py:131: error: Argument 1 of "__add__" is incompatible with supertype "int"; supertype defines the argument type as "int"  [override]
nncf/api/compression.py:131: note: This violates the Liskov substitution principle
nncf/api/compression.py:131: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```
Shortly, unsafe to override a method with a more specific argument type and unsafe to override a method with a more general return type.